ethereum_history_api/oracles/src/ethereum/transaction.ts

```diff
@@ -0,0 +1,102 @@
+import {
+  AccessList,
+  Hex,
+  Transaction,
+  TransactionBase,
+  TransactionEIP1559,
+  TransactionEIP2930,
+  TransactionEIP4844,
+  TransactionLegacy,
+  concatHex,
+  toRlp
+} from 'viem';
+import { RecursiveArray, txTypeToHex } from './receipt.js';
+import { encodeField } from '../noir/oracles/common/encode.js';
+
+export function accessListToRlpFields(accessList: AccessList): RecursiveArray<Hex> {
+  return accessList.map(({ address, storageKeys }) => [address, storageKeys]);
+}
+
+type TxBase = Omit<TransactionBase, 'yParity'>;
+
+export function signatureToRlpFields(baseTx: TxBase): Hex[] {
+  return [encodeField(baseTx.v), baseTx.r, baseTx.s];
+}
+
+export function toValueInputTxToRlpFields(tx: TxBase): Hex[] {
```

Comment:
I also don't like this name, but extracting those middle fields as a function - makes other functions much more readable as we see - what's different between them. Please suggest a better name

Reply:
Wrap in namespace/class/struct?
e.g.
encode = TransactionRlpEncoder(tx)
encode.signature()

than static function to create an instance:
```ts
TransactionRlpEncoder.encode()
```

sth like this

Reply:
Could you please elaborate why? I feel like most of the modules in our code are lists of functions instead of classes and that's been our style up to this point. It makes sense to have a class when you have some properties, but here all the functions are still static. I agree - that having them within a class - allows to reuse class name as a context and have shorter function names. Is this what you meant here? Either way, changed it. Please take another look and check if I guessed your intention correctly.